### PR TITLE
fix requests for files in group folders to be found

### LINKED ISSUE #8: Browsing issue
Hi,

The task browses a single user folder and then stops. it does'nt browse the __groupfolders folder.

### PR DESCRIPTION
**Pull request by mgallien**
Fix https://github.com/nextcloud-gmbh/mtime_fixer_tool_kit/issues/8

Signed-off-by: Matthieu Gallien <matthieu.gallien@nextcloud.com>

### CODE REVIEW

File: scanGroupFolders.sh [hunk 1]
```diff
@@ -0,0 +1,6 @@
+#! /bin/bash
+
+for i in $(php ./occ groupfolders:list --output=json_pretty | grep '"id":' | sed "s/.*id\": //" | sed "s/,//"); do
+	php ./occ groupfolders:scan $i
+done
+
```
Comment (artonge) [inline scanGroupFolders.sh]: Is it useful to keep this file?

File: solvable_files.sh [hunk 2]
```diff
@@ -28,6 +28,7 @@ function correct_mtime() {
 	filepath="$1"
 	relative_filepath="${filepath/#$data_dir\//}"
 	mtime_on_fs="$(stat -c '%Y' "$filepath")"
+	data_dir_with_trailing_slash="$data_dir/"
```
Comment (artonge) [inline solvable_files.sh:31]: Let's remove that variable and add the `/` in the query.
```suggestion
```

File: solvable_files.sh [hunk 3]
```diff
@@ -53,7 +55,7 @@ function correct_mtime() {
 					--execute="\
 						SELECT mtime
 						FROM oc_storages JOIN oc_filecache ON oc_storages.numeric_id = oc_filecache.storage \
-						WHERE oc_storages.id='local::$data_dir' AND oc_filecache.path=FROM_BASE64('$base64_relative_filepath_without_username')" \
+						WHERE oc_storages.id='local::$data_dir_with_trailing_slash' AND oc_filecache.path=FROM_BASE64('$base64_relative_filepath')" \
```
Comment (artonge) [inline solvable_files.sh]:
```suggestion
						WHERE oc_storages.id='local::$data_dir/' AND oc_filecache.path=FROM_BASE64('$base64_relative_filepath')" \
```

File: solvable_files.sh [hunk 4]
```diff
@@ -67,7 +69,7 @@ function correct_mtime() {
 					--command="\
 						SELECT mtime
 						FROM oc_storages JOIN oc_filecache ON oc_storages.numeric_id = oc_filecache.storage \
-						WHERE oc_storages.id='local::$data_dir' AND oc_filecache.path='CONVERT_FROM(DECODE($base64_relative_filepath_without_username, 'base64'), 'UTF-8')'"
+						WHERE oc_storages.id='local::$data_dir_with_trailing_slash' AND oc_filecache.path='CONVERT_FROM(DECODE($base64_relative_filepath, 'base64'), 'UTF-8')'"
```
Comment (artonge) [inline solvable_files.sh]:
```suggestion
						WHERE oc_storages.id='local::$data_dir/ AND oc_filecache.path='CONVERT_FROM(DECODE($base64_relative_filepath, 'base64'), 'UTF-8')'"
```

**Review by artonge — CHANGES_REQUESTED** [review]
Can you also change every `data_dir="$1"` to `data_dir="$(realpath "$1")"` ?

File: scanGroupFolders.sh (hunk 1 above)
Comment (artonge) [reply scanGroupFolders.sh]: Ok I understand why it is necessary.

**Review by artonge — APPROVED** [review]

**Comment by mgallien** [comment]
thanks @artonge

**Pull request merged**